src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequestHelper.cs

```diff
@@ -41,70 +52,104 @@ internal static async Task<MsalTokenResponse> RefreshAccessTokenAsync(MsalRefres
+        {
+            refreshOnWithJitter = GetRefreshOnWithJitter(oldAccessToken);
+            if (refreshOnWithJitter.HasValue && refreshOnWithJitter.Value < DateTimeOffset.UtcNow)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Fire and forget the fetch action on a background thread.
+        /// Do not change to Task and do not await it.
+        /// </summary>
+        internal static void ProcessFetchInBackground(
+            MsalAccessTokenCacheItem oldAccessToken,
+            Func<Task<AuthenticationResult>> fetchAction,
+            ICoreLogger logger)
+        {
+            _ = Task.Run(async () =>
+            {
+                try
+                {
+                    await fetchAction().ConfigureAwait(false);
+                }
+                catch (MsalServiceException ex)
+                {
+                    string logMsg = $"Background fetch failed with MsalServiceException. Is AAD down? { ex.IsAadUnavailable()}";
+                    if (ex.StatusCode == 400)
+                    {
+                        logger.ErrorPiiWithPrefix(ex, logMsg);
+                    }
+                    else
+                    {
+                        logger.WarningPiiWithPrefix(ex, logMsg);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    string logMsg = $"Background fetch failed with exception.";
+                    logger.WarningPiiWithPrefix(ex, logMsg);
+                }
+            });
+        }
+
+        private static Random s_random = new Random();
+        private static DateTimeOffset? GetRefreshOnWithJitter(MsalAccessTokenCacheItem msalAccessTokenCacheItem)
+        {
+            if (msalAccessTokenCacheItem.RefreshOn.HasValue)
+            {
+                int jitter = s_random.Next(-Constants.DefaultJitterRangeInSeconds, Constants.DefaultJitterRangeInSeconds);
+                var refreshOnWithJitter = msalAccessTokenCacheItem.RefreshOn.Value + TimeSpan.FromSeconds(jitter);
+                return refreshOnWithJitter;
+            }
+
+            
+            return null;           
+        }
+
+        private static string ExtractEnrollmentId(string enrollmentIDs)
+        {
+            // This method extracts EnrollmentId from EnrollmentIDs that correspoond ot the value in the keychain
+            // expected input format is as follows
+            //{
+            //    "enrollment_ids" : [
+            //      {
+            //          "home_account_id" : "29ba73b9-5cbb-478b-a67d-1e8cf7dd425b.7257a09f-53cc-4a91-aca8-0cb6713642a5",
+            //          "tid" : "7257a09f-53cc-4a91-aca8-0cb6713642a5",
+            //          "user_id" : "IDLAB20TrueMAMCA@msidlab20.onmicrosoft.com",
+            //          "oid" : "29ba73b9-5cbb-478b-a67d-1e8cf7dd425b",
+            //          "enrollment_id" : "02460f14-a7e5-4458-9ab1-789d97114aae"
```

Comment:
I would remove the values or put fake guids, if they not already are.

Reply:
Fakes